torchmetrics/functional/text/infolm.py

```diff
@@ -0,0 +1,611 @@
+import os
+from enum import unique
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import torch
+from torch import Tensor
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+from typing_extensions import Literal
+
+from torchmetrics.functional.text.helper_embedding_metric import (
+    TokenizedDataset,
+    _get_progress_bar,
+    _input_data_collator,
+    _load_tokenizer_and_model,
+)
+from torchmetrics.utilities.enums import EnumStr
+from torchmetrics.utilities.imports import _TRANSFORMERS_AVAILABLE
+
+if _TRANSFORMERS_AVAILABLE:
+    from transformers import PreTrainedModel, PreTrainedTokenizerBase
+else:
+    __doctest_skip__ = ["infolm"]
+
+
+_ALLOWED_INFORMATION_MEASURE = (
+    "kl_divergence",
+    "alpha_divergence",
+    "beta_divergence",
+    "ab_divergence" "renyi_divergence",
+    "l1_distance",
+    "l2_distance",
+    "l_infinity_distance",
+    "fisher_rao_distance",
+)
+
+
+_ALLOWED_INFORMATION_MEASURE_LITERAL = Literal[
```

Comment:
this seems to be using the same list, so let's check if we can do something like 
```py
_ALLOWED_INFORMATION_MEASURE_LITERAL = Literal[*_ALLOWED_INFORMATION_MEASURE]
```

Reply:
Actually, this formulation seems to be possible as of `python>=3.11`. But I tried to simplify that as much as possible and add a test verifying that the list of allowed measure is equivalent to its type hint counterpart.